src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -1080,6 +1086,26 @@ protected bool CheckBooleanColumnType(DType type, TexlNode arg, IErrorContainer
             return CheckColumnType(type, arg, DType.Boolean, errors, TexlStrings.ErrInvalidSchemaNeedColorCol_Col, ref nodeToCoercedTypeMap);
         }
 
+        protected DType NumDecReturnType(CheckTypesContext context, DType argType)
+        {
+            IEnumerable<TypedName> columns;
+
+            if (argType.IsTable && (columns = argType.GetNames(DPath.Root)).Count() == 1)
+            {
+                argType = columns.Single().Type;
+            }
+
+            // Decimal TODO: Is this the right thing to do for deferred and unknown?

```

Comment:
>Decimal TODO [](http://example.com/codeflow?start=15&length=12)

definitely for deferred, but I'm not entirely sure what this is trying to do otherwise.